test/grammar_test.rb

```diff
@@ -9,37 +9,37 @@ def test_comment
 
   def test_key
     match = TomlRB::Document.parse('bad_key-', root: :key)
-    assert_equal('bad_key-', match.value)
+    assert_equal('bad_key-', match.value.first)
```

Comment:
I am a little concerned about this, because it is breaking backwards compatibility.

Reply:
Or I am getting something wrong from this changes? 🤔

Reply:
To simplify the implementation of the dotted key, single key is regarded as a dotted key with one element(Table key is the same).

Certainly, if backwards compatibility is important,
I think it is necessary to separate single key and dotted key.

Reply:
Well, I would love to keep backwards compatibility but I also understand that is not a requirement since TOML is not necessary backwards compatible until it reaches 1.0.

So I'm happy to leave this as is, until someone reports a bug.